src/onnx/onnx.cpp

```diff
@@ -750,15 +750,20 @@ struct onnx_parser
         {
             auto names = attributes.at("activations").strings();
             vec_names.clear();
-            for_each(names.begin(), names.end(), [&](auto& fn) { vec_names.push_back(fn); });
+            vec_names.resize(names.size());
+            std::transform(
+                names.begin(), names.end(), vec_names.begin(), [](auto& str) { return str; });
         }
 
-        for_each(vec_names.begin(), vec_names.end(), [&](auto& fn) {
-            if(map_actv_funcs.count(fn) == 0)
-            {
-                MIGRAPHX_THROW("RNN: activation function " + std::string(fn) + " not supported");
-            }
-        });
+        if (std::any_of(vec_names.begin(), vec_names.end(), [&](auto& name) {
+            return (map_actv_funcs.count(name) == 0);
+        }))
+        {
+            auto name_it = std::find_if(vec_names.begin(), vec_names.end(), [&] (auto &name) {
```

Comment:
Sorry, I guess `any_of` was a bad choice because the the error message. Actually we can do `find_if` first, and then check if it is not found and print the message. This way we dont have to search twice.